lib/data_update_scripts/20210423155327_migrate_relevant_fields_from_users_to_users_settings.rb

```diff
@@ -1,86 +1,9 @@
-                    WHEN config_theme='minimal_light_theme' THEN 1
-                    WHEN config_theme='night_theme' THEN 2
-                    WHEN config_theme='pink_theme' THEN 3
-                    WHEN config_theme='ten_x_hacker_theme' THEN 4
-                    ELSE 0
-                END
-                config_theme,
-                COALESCE(display_announcements, true),
-                COALESCE((data ->> 'display_email_on_profile')::boolean, false) as display_email_on_profile,
-                COALESCE(display_sponsors, true),
-                CASE WHEN editor_version='v2' THEN 0
-                      WHEN editor_version='v1' THEN 1
-                      ELSE 0
-                END
-                editor_version,
-                experience_level,
-                COALESCE(feed_mark_canonical, false),
-                COALESCE(feed_referential_link, true),
-                feed_url,
-                inbox_guidelines,
-                CASE WHEN inbox_type='private' THEN 0
-                      WHEN inbox_type='open' THEN 1
-                      ELSE 0
-                END
-                inbox_type,
-                COALESCE(permit_adjacent_sponsors, true),
-                users.id AS user_id,
-                NOW(),
-                NOW()
-              FROM users
-              JOIN profiles
-                ON profiles.user_id = users.id
-            )
-            INSERT INTO users_settings (brand_color1, brand_color2, config_font, config_navbar, config_theme, display_announcements, display_email_on_profile, display_sponsors, editor_version, experience_level, feed_mark_canonical, feed_referential_link, feed_url, inbox_guidelines, inbox_type, permit_adjacent_sponsors, user_id, created_at, updated_at)
-              SELECT * FROM settings_data
-              ON CONFLICT (user_id) DO UPDATE
-                SET brand_color1 = EXCLUDED.brand_color1,
-                    brand_color2 = EXCLUDED.brand_color2,
-                    config_font = EXCLUDED.config_font,
-                    config_navbar = EXCLUDED.config_navbar,
-                    config_theme = EXCLUDED.config_theme,
-                    display_announcements = EXCLUDED.display_announcements,
-                    display_email_on_profile = EXCLUDED.display_email_on_profile,
-                    display_sponsors = EXCLUDED.display_sponsors,
-                    editor_version = EXCLUDED.editor_version,
-                    experience_level = EXCLUDED.experience_level,
-                    feed_mark_canonical = EXCLUDED.feed_mark_canonical,
-                    feed_referential_link = EXCLUDED.feed_referential_link,
-                    feed_url = EXCLUDED.feed_url,
-                    inbox_guidelines = EXCLUDED.inbox_guidelines,
-                    inbox_type = EXCLUDED.inbox_type,
-                    permit_adjacent_sponsors = EXCLUDED.permit_adjacent_sponsors,
-                    updated_at = NOW();
-          SQL
-        )
+      User.ids.each do |id|
```

Comment:
I'm a little worried that the sheer size of the number of jobs this would create would drown Sidekiq/Redis. Am I wrong to assume that?

Reply:
If our queue can't handle 600k jobs in the default priority than we need to regroup :P 
An alternative could be to have each job take a batch of ids but then it's going to be a PITA when batch jobs start failing because one of the users has corrupted data and you don't know which one, so you have to add in depth error handling and so on and so forth.

I think the strategy of one job per user is solid. 

Can't we just raise the number of workers temporarily if we're worried?

Reply:
Out of curiosity do we know on average how many jobs can our queue handle? Or what's the max that we have tested it with before?